src/AssetManager.php

```diff
@@ -52,22 +103,85 @@ public function markAsLoaded($asset)
      * @param  string  $asset
      * @return bool
      */
-    public function isLoaded($asset)
+    public function isLoaded(string $asset): bool
     {
-        if (in_array($asset, $this->loaded)) {
-            return true;
-        }
-
-        return false;
+        return in_array($asset, $this->loaded);
     }
 
     /**
      * Returns the current loaded assets on app lifecycle.
      *
      * @return array
      */
-    public function loaded()
+    public function loaded(): array
     {
         return $this->loaded;
     }
+
+    /**
+     * Localize a CDN asset.
+     *
+     * @param  string  $asset
+     * @return void
+     */
+    public function basset(string $asset, bool $output = true, array $attributes = [], string $type = null): string
```

Comment:
What if the second parameter here... `output`... wasn't just a `boolean` but also a `string`? So that you can pass if it should output (yes, no or file path including extension). Couldn't that solve the Google Fonts problem, for example?

```diff
-@basset('https://fonts.googleapis.com/css2?family=Source+Sans+Pro:ital,wght@0,400;0,600;0,700;1,400&display=swap')
+@basset('https://fonts.googleapis.com/css2?family=Source+Sans+Pro:ital,wght@0,400;0,600;0,700;1,400&display=swap', 'fonts.googleapis.com/source_sans_pro.css')
```

Then we know this file would be saved with the `.css` extension, right?

Reply:
This might ALSO solve the problem of not being able to load the files if the CDN URL changes. For example, the unpkg URLs with `1.*.*`. If we give specify the name... 🐶 will know _exactly_ where to look, right?

Reply:
Of course, if we can solve this by looking at the final file name... and final mime type... we should do that, not bloat most definitions with a name.